src/Surfnet/StepupGateway/GatewayBundle/Tests/Service/Gateway/ConsumeAssertionServiceTest.php

```diff
@@ -447,6 +448,179 @@ public function it_should_throw_an_exception_when_the_in_respone_to_is_invalid_w
+',
+        ], $this->getSessionData('attributes'));
+    }
+
+    public function test_it_rejects_nameidless_responses()
+    {
+        // When the IdP did not add a subject nameid and also skips on the internal-collabPersonId. The assertion
+        // is a 'valid' message according to our SAML processor (SAML2 library). But we can't process it, in
+        // Gateway due to the lack of an identity id to match SF tokens with.
+        $samlResponseXml = '<samlp:Response xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" ID="CORTO1111111111222222222233333333334444444444" Version="2.0" IssueInstant="2014-10-22T11:09:59Z" Destination="https://gateway.org/acs" InResponseTo="_mocked_generated_id">
+    <saml:Issuer>https://idp.edu/metadata</saml:Issuer>
+    <samlp:Status>
+        <samlp:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success"/>
+    </samlp:Status>
+    <saml:Assertion xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xs="http://www.w3.org/001/XMLSchema" ID="pfxaa30421c-e921-072d-d75a-decd1a4ee918" Version="2.0" IssueInstant="2014-10-22T11:09:59Z">
+        <saml:Issuer>https://idp.edu/metadata</saml:Issuer>
+        <saml:Subject>
+            
+            <saml:SubjectConfirmation Method="urn:oasis:names:tc:SAML:2.0:cm:bearer">
+                <saml:SubjectConfirmationData IssueInstant="2014-10-22T11:09:59Z" Recipient="https://gateway.org/acs" InResponseTo="_mocked_generated_id"/>
```

Comment:
I did not sign the assertion on this response. I felt that did not add to the situation under test here.